Automattic/jetpack
# Introduce Jetpack Newsletter in the recommendations flow

## Proposed changes:

This PR adds a new `/recommendation/#newsletter` path for the Recommendations

<img width="1062" alt="Screenshot 2023-07-18 at 14 55 37" src="https://github.com/Automattic/jetpack/assets/4077604/5d1ed324-18d5-4b00-85c6-12989a763a9f">

It also renames Subscriptions => Newsletter in the settings page

<img width="1121" alt="Screenshot 2023-07-18 at 14 23 26" src="https://github.com/Automattic/jetpack/assets/4077604/2f100cc9-26a5-4ed4-9090-ca911d1132d9">

## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

My PR adds *x* and *y*.

## Testing instructions
- If running Jetpack in development mode, clear the options at the bottom of the page
- Step through the "Recommendations" using Jetpack Free (i.e. not a paid plan)
- You should see the Newsletter recommendation
- Enable it, it should show up on the summary 
- Navigate directly to the card `wp-admin/admin.php?page=jetpack#/recommendations/newsletter` it should show the status of the feature.

Copy is (and design image) still to be finalised - do not merge until later.
